### ISSUE TITLE
Feature Request: Copy to clipboard buttons on code blocks

### ISSUE THREAD

**Issue by damon314159** [TheOdinProject/theodinproject#4416]
### Checks

- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [X] The title of this issue follows the `Feature Request: brief description of feature request` format, e.g. `Feature Request: Add a dark mode to the website`
- [ ] Would you like to work on this issue?

### Description of the Feature Request

I was doing the setup for a new machine last night and it occurred to me that TOP doesnt have those QoL copy icons on code blocks that copy the code block to your clipboard, so you have to select the whole command yourself. Could be a nice feature to include
See here an example on an mdn reference page:
![image](https://github.com/TheOdinProject/theodinproject/assets/126731021/7e1859c5-43c7-42e0-b7f2-f44244e4e43c)

### Acceptance criteria

- [ ] a copy icon that copies a code block into user's clipboard is added

**Comment by KevinMulhern** [comment]
Thanks for the suggestion @damon314159, I agree it makes a lot of sense having the snippets on installations copyable. 

I remember this coming up before, either on Discord or on Github, and we were reluctant to add it to every lesson.

IIRC adding a copy button would make it easier to copy and paste, and we wanted to encourage leaners to type out the code snippets in the programming lessons. Theres value in typing it out, making mistakes and debugging them. It builds a lot of muscle memory and gives learners a better feel of how the code works.

_But_... that might be old school kind of thinking. I'd appreciate some thoughts on this. I'm happy to go with whatever the consensus is.

In short: it would be easier to add the clipboard button site wide for every lesson. But if that isn't something we want, we'll need to figure out a way to enable it for specific (installation) lessons - still very doable.

**Comment by Asartea** [comment]
I think there is still added value in typing even bash commands, as it forces you to think about the structure of a command.

**Comment by damon314159** [comment]
I think if a person intends to copy it, they're going to select and copy it regardless. This would only be a QoL change in that regard.

I can see the argument for only applying it to bash - that makes sense to me.

Perhaps a note box when the idea of coding along is introduced that typing code yourself builds memory

**Comment by KevinMulhern** [comment]
Targeting bash only blocks is a great idea to get this started.

I've been doing a little research to hopefully help whoever picks this up. Prism have a copy-to-clipboard plugin we can use: https://prismjs.com/plugins/copy-to-clipboard/

We include new Prism plugins in our [.babelrc file](https://github.com/TheOdinProject/theodinproject/blob/main/.babelrc#L5), we'll need something like this:
 ```
 "plugins": ["line-numbers", "toolbar", "copy-to-clipboard"]
```

After that, the rest should hopefully be manageable with CSS 🤞

**Comment by Asartea** [comment]
I would be willing to take this on. Playing around with it its relatively easy to hide the button using CSS for anything that isn't bash using a class selector and display:none:

```css
pre.highlight:not(.language-bash) + .toolbar {
  display: none;
}
```

The one annoying thing with that is that we are still running the code in all cases, just hiding the output of it

**Comment by KevinMulhern** [comment]
Thanks @Asartea, I've assigned to you.

**Comment by Asartea** [comment]
Okay stupid question: how do I get CSS changes to show up? I inserted the above line into [prism-theme.css](https://github.com/TheOdinProject/theodinproject/blob/main/app/assets/stylesheets/stylesheets/prism-theme.css), but running the app locally doesn't actually seem to reflect that change. I tried restarting `foreman`, as well as clobbering the CSS (via `rails css:clobber`)

**Comment by KevinMulhern** [comment]
That _should_ have done it. But try running `rails assets:clobber`, I think its a bit more thorough.

**Comment by Asartea** [comment]
Ah that did it
